assets/app/scripts/controllers/pod.js

```diff
@@ -26,6 +26,19 @@ angular.module('openshiftConsole')
       }
     ];
 
+    $scope.selectedTab = {};
+    if ($routeParams.tab) {
+      $scope.selectedTab[$routeParams.tab] = true;
+    }
+
+    $scope.$watch('selectedTab', function() {
+      angular.forEach($scope.selectedTab, function(selected, tab) {
+        if (selected) {
+          $location.search('tab', tab);
+        }
+      });
+    }, true);
+
     var watches = [];
 
     project.get($routeParams.project).then(function(resp) {
```

Comment:
Quick suggestion for using project service, add `_.spread` to the `then()` so we can name the args:

```javascript
project
  .get($routeParams.project)
  .then(_.spread(function(project, context) {
      // yay, don't need array syntax!
  }));
```
And rather than passing `$scope` to the `DataService.get("pods"`, we can just pass along `context`:

```javascript
DataService
  .get("pods", $routeParams.pod, context)
  .then( /* more stuff... */);
```
Which is nice because `$scope` will then only be 'travelling' forward to the view (not back into services), and we don't have to worry about `projectPromise`.